include/flecs.h

```diff
@@ -4157,6 +4157,47 @@ const ecs_type_info_t* ecs_get_type_info(
     const ecs_world_t *world,
     ecs_id_t id);
 
+
+/**
+ * @defgroup illegal_hooks Illegal lifecycle hooks
+ * Functions to mark specific lifecycle hooks as illegal
+ *
+ * @{
+ */
+
+/* Illegal constructor handler and constant */
+FLECS_API
+ecs_xtor_t ecs_ctor_illegal_(void);
+#define ECS_CTOR_ILLEGAL (ecs_ctor_illegal_())
```

Comment:
Why are these macro's necessary? Can't we just use `ecs_ctor_illegal_` directly?

Reply:
I thought the same. My first version was like that. Then msys tests failed. Apparently, in Windows for DLLs, an exported function may be a proxy function to the real function inside, so comparisons failed and the test to detect the illegal hook failed. The macro is there to give a sense it is like a constant.